src/tbp/monty/frameworks/models/buffer.py

```diff
@@ -608,6 +608,15 @@ def register(
         else:
             raise ValueError(f"Invalid encoder: {encoder}")
 
+    @classmethod
+    def unregister(cls, obj_type: type) -> None:
```

Comment:
Does this exist only for the tests, or do we expect to use this elsewhere? If it's just for the tests, then maybe my suggestion below for how to reset the BufferEncoder class in the tests is a better approach.

Reply:
Yes, this emerged from testing needs.

Reply:
I'm a little suspicious of adding functionality just for the sake of tests, but it's probably fine here.

Reply:
I think the approach of having it shared on the class makes it more flexible, and to me that's worth more than a little discomfort around an extra method for the tests.